Content.Server/GameObjects/Components/Items/Storage/EntityStorageComponent.cs

```diff
@@ -331,25 +334,43 @@ public bool CanInsert(IEntity entity)
 
         async Task<bool> IInteractUsing.InteractUsing(InteractUsingEventArgs eventArgs)
         {
-
             if (Open)
+            {
+                _beingWelded = false;
                 return false;
+            }
 
             if (!CanWeldShut)
+            {
+                _beingWelded = false;
                 return false;
+            }
 
             if (Contents.Contains(eventArgs.User))
             {
+                _beingWelded = false;
                 Owner.PopupMessage(eventArgs.User, Loc.GetString("It's too Cramped!"));
                 return false;
             }
 
-            if (!eventArgs.Using.TryGetComponent(out WelderComponent tool))
+            if (!eventArgs.Using.TryGetComponent(out WelderComponent? tool) || !tool.WelderLit)
+            {
+                _beingWelded = false;
+                return false;
+            }
+
+            if (_beingWelded)
                 return false;
 
+            _beingWelded = true;
```

Comment:
`_beingWelded` should be checked at the start of the method, otherwise you could get someone trying to weld in a way that doesn't work resetting `_beingWelded` to false even if it's currently being welded.

Reply:
Hmm I should probably move the checks into the DoAfter and just do the flat _beingWelded up front right.